### DISCUSSION TITLE
GraphQL

### DISCUSSION THREAD

**Discussion by chrisshaw** [Budibase/budibase#1402] [Ideas]
This is to second the requests made by https://github.com/Budibase/budibase/discussions/891#discussioncomment-270783 and https://github.com/Budibase/budibase/discussions/891#discussioncomment-402184 and make a squeakier wheel.

I appreciate that with HTTP we can replicate this ourselves. But it'd be much, much faster and safer to have it auto-generated from introspection as we have a fairly complex schema (don't we all).

**Comment by shogunpurple** [comment]
Thanks for raising @chrisshaw - we have created an issue for this one, and should implement it over the coming months. 
I really like the suggestion about using GQL introspection to identify the schema, and allowing the user to configure their schema from there.

https://github.com/Budibase/budibase/issues/1405

Are there any other features of a graphQL integration that would make this a great experience for you?

**Reply by chrisshaw** [reply]
**OAuth.** We're using Auth0 with client credentials and passing the JWT as a header to GraphQL. (I think someone else mentioned this elsewhere.)

---

Nice to haves for which I'm not really expecting the effort-impact tradeoff to be there, so feelings won't be hurt if they never come:

**AuthN layer.** We're doing this in the GraphQL server but it'd be awesome if we could use Budibase's built-in AuthN so it's all in one place if we also use built-in tables

> (although truth be told I'm not sure how BB's Data authorization is supposed to work; as long as I stick it in the UI, a Basic user can still read and write to tables that are supposed to be gated to Admin. Is this a bug or am I just dumb?)

I can easily see how this would be out of scope, especially given we're involved in field-level ABAC.

 **Auto-filling CRUD queries as CRUD.** (This is true across external sources really; just another attempt at getting external data sources closer to internal.) We have a number of resourceful CRUD operations and then a growing handful of workflow operations. It'd be nice if we didn't have to go in and specify each of these operations' functions appropriately. In my very specific use case there's a couple clear patterns to follow but I'm not sure how else you'd make this work.

**Comment by chrisshaw** [comment]
I saw this is stale, I'll let you decide how the Issue workflow should work: https://github.com/Budibase/budibase/issues/1405

In timely news, ToolJet just released their GraphQL integration: https://github.com/ToolJet/ToolJet/releases/tag/v0.5.11

**Reply by shogunpurple** [reply]
Thanks @chrisshaw - I gave it a little bump there to avoid it going stale. Thanks very much for your patience. This one is still on our roadmap and we will make sure to let the community know upon its release!
